### ISSUE TITLE
Update consumers of typed array API to use CreateWith argument

### ISSUE THREAD

**Issue by jdm** [servo/servo#15553]
https://github.com/servo/rust-mozjs/issues/336 modified the typed array API constructors to accept a single argument instead of a optional slice as well as a length. We should upgrade the version of rust-mozjs in use in Servo and update the code that uses this API. Start with `./mach cargo-update -p js` and then fix the resulting errors that will appear when running `./mach build`.

No need to run any tests for this change; if it builds, it's good enough for a PR.

**Comment by jdm** [comment]
Would you like to work on this, @abhiQmar?

**Comment by zploskey** [comment]
I can give this a go if that's alright.
@highfive: assign me

**Comment by jdm** [comment]
@zploskey Actually I am waiting on @abhiQmar to give first refusal, sorry.

**Comment by zploskey** [comment]
@jdm Ok, no worries.

**Comment by abhiQmar** [comment]
@jdm Yes, I want to work on this. Thanks for the opportunity Josh :)
@highfive: assign me

**Comment by Ms2ger** [comment]
This is blocking me, I'm fixing.

**Comment by jdm** [comment]
Sorry @abhiQmar. I'll look for another issue that you could work on.
